common/formats/src/format_csv.rs

```diff
@@ -311,4 +353,126 @@ impl InputFormat for CsvInputFormat {
         }
         Ok(0)
     }
+
+    fn read_row_num(&self, state: &mut Box<dyn InputState>) -> Result<usize> {
+        let state = state.as_any().downcast_mut::<CsvInputState>().unwrap();
+        Ok(state.accepted_rows)
+    }
+}
+
+#[allow(clippy::format_push_string)]
+impl FormatDiagnostic for CsvInputFormat {
+    fn deserialize_field_and_print_diagnositc_info(
```

Comment:
I think this function has less relationship with the format. So we need to implement it just once.

Reply:
make sense